### PR TITLE
nrf_modem: add dect-phy binaries

### PR DESCRIPTION
**Pull request by eivindj-nordic**
Add Dect NR+ PHY modem library binaries to nrfxlib.

### CODE REVIEW

File: nrf_modem/dect_phy/include/nrf_modem_dect_phy.h [hunk 1]
```diff
@@ -0,0 +1,1190 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA. All Rights Reserved.
```
Comment (MirkoCovizzi) [inline nrf_modem/dect_phy/include/nrf_modem_dect_phy.h:2]:
```suggestion
 * Copyright (c) 2024 Nordic Semiconductor ASA. All Rights Reserved.
```

File: nrf_modem/Kconfig [hunk 2]
```diff
@@ -15,12 +15,15 @@ choice NRF_MODEM_BUILD_STRATEGY
 	depends on NRF_MODEM
 	prompt "Build strategy"
 
-config NRF_MODEM_LINK_BINARY
-	bool "Link binary"
+config NRF_MODEM_LINK_BINARY_LTE
+	bool "Link LTE binary"
+
+config NRF_MODEM_LINK_BINARY_DECT_PHY
+	bool "Link DECT binary"
```
Comment (lemrey) [inline nrf_modem/Kconfig:22]: Perhaps `DECT PHY` is more correct here?

**Review by lemrey — COMMENTED** [review]
Looks good, structure is correct -- just a minor comment.
We can go ahead and call it `cellular`.

File: nrf_modem/Kconfig (hunk 2 above)
Comment (anhmolt) [reply nrf_modem/Kconfig]: Renamed in the second commit (lte->cellular rename commit). Maybe it should be in the first.

**Review by anhmolt — APPROVED** [review]

**Comment by lemrey** [comment]
Could we avoid adding the binaries & headers here? We only need the `CMakeLists.txt` and `Kconfig` updates.

**Comment by eivindj-nordic** [comment]
> Could we avoid adding the binaries & headers here? We only need the `CMakeLists.txt` and `Kconfig` updates.

I would like to keep it to be able to build NCS (dect) samples on the integration branches without building from source.

**Review by lemrey — APPROVED** [review]

**Pull request merged**